Bocharov.egor/S2

LuckyBichar opened the pull request:
Вроде всё сделал
Тесты: https://github.com/LuckyBichar/spbspu-labs-2025-aads-a/actions/runs/14128833840
![image](https://github.com/user-attachments/assets/a7889d01-c8a7-43e2-9e7d-71e1cf36b681)

LuckyBichar committed:
initial commit s2

LuckyBichar committed:
array and test-main

LuckyBichar committed:
stack

LuckyBichar committed:
queue

LuckyBichar committed:
actions

LuckyBichar committed:
main

LuckyBichar committed:
fix

LuckyBichar committed:
test-array

LuckyBichar committed:
test-queue and test-stack

Volgarenok commented:
Баллы то какие

LuckyBichar committed:
fix node.hpp

LuckyBichar committed:
fix queue and stack

LuckyBichar committed:
fix array

LuckyBichar committed:
fix tests

LuckyBichar committed:
fix

LuckyBichar committed:
Merge remote-tracking branch 'origin/master' into bocharov.egor/S2

Volgarenok commented on bocharov.egor/S2/test-stack.cpp:
```diff
@@ -0,0 +1,52 @@
+#include <boost/test/tools/interface.hpp>
+#include <boost/test/unit_test.hpp>
+#include <boost/test/unit_test_suite.hpp>
+#include <sys/types.h>
```
Причём тут sys?..

LuckyBichar replied:
упс

Volgarenok commented on bocharov.egor/S2/queue.hpp:
```diff
@@ -0,0 +1,90 @@
+    T & front();
+    const T & front() const;
+    T & back();
+    const T & back() const;
```
У очереди вы должны использовать только front

LuckyBichar replied:
исправил

Volgarenok commented on bocharov.egor/S2/main.cpp:
```diff
@@ -0,0 +1,29 @@
+    bocharov::outputResults(std::cout, results) << "\n";
+  }
+  catch (std::exception & e)
+  {
```
Дурка

LuckyBichar replied:
согласен

Volgarenok commented on bocharov.egor/S2/main.cpp:
```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include "actions.hpp"
+
+int main(int argc, char * argv[])
```
using namespace

LuckyBichar replied:
сделал

Volgarenok commented on bocharov.egor/S2/array.hpp:
```diff
@@ -0,0 +1,240 @@
+
+  private:
+
+    T ** data_;
```
Я не очень понимаю с какой целью делать указатель на указатель? Почему нельзя хранить в массиве сами данные?

LuckyBichar replied:
исправил

Volgarenok commented on bocharov.egor/S2/array.hpp:
```diff
@@ -0,0 +1,240 @@
+  template< typename T >
+  void Array< T >::clear() noexcept
+  {
+    while(!empty())
```
CG

LuckyBichar replied:
исправил

Volgarenok commented on bocharov.egor/S2/actions.hpp:
```diff
@@ -0,0 +1,21 @@
+#include "stack.hpp"
+#include <iostream>
+#include <string>
+
```
CG

LuckyBichar replied:
исправил

Volgarenok replied:
Всё ещё CG

LuckyBichar replied:
исправил

Volgarenok commented on bocharov.egor/S2/actions.cpp:
```diff
@@ -0,0 +1,246 @@
+      {
+        try
+        {
+          long long int number = stoll(partPostfix);
```
CG

LuckyBichar replied:
готово

LuckyBichar committed:
fix

LuckyBichar committed:
fix fix

LuckyBichar committed:
fix fix fix

LuckyBichar committed:
fix array

LuckyBichar committed:
Merge remote-tracking branch 'origin/master' into bocharov.egor/S2

LuckyBichar committed:
Merge remote-tracking branch 'origin/master' into bocharov.egor/S2

LuckyBichar committed:
modified files

LuckyBichar committed:
delete

LuckyBichar committed:
common

LuckyBichar committed:
Merge remote-tracking branch 'origin/master' into bocharov.egor/S2

Volgarenok commented on bocharov.egor/common/array.hpp:
```diff
@@ -0,0 +1,232 @@
+    begin_(0)
+  {
+    data_ = new T [capacity_];
+  }
```
А почему нельзя в списке инициализации это сделать?

LuckyBichar replied:
Готово

Volgarenok commented on bocharov.egor/common/array.hpp:
```diff
@@ -0,0 +1,232 @@
+  Array< T >::~Array()
+  {
+    delete[] data_;
+    capacity_ = 0;
```
А если 2 поставить, а не 0, то что будет?

LuckyBichar replied:
убрал

LuckyBichar replied:
йоу

Volgarenok commented on bocharov.egor/common/array.hpp:
```diff
@@ -0,0 +1,232 @@
+    std::swap(begin_, arr.begin_);
+  }
+
+  template< typename T >
+  Array< T >::Array():
```
Другой конструктор здесь нельзя переиспользовать?

LuckyBichar replied:
вроде готово

LuckyBichar committed:
⣿⣿⣿⣿⣿⢿⠿⣛⣯⢭⣶⣲⢶⣖⣮⣬⣜⡨⣛⢿⣿⣿⣿⣿
⣿⡟⣭⣶⣶⣶⣿⣿⣿⣿⣾⣿⣽⣿⣾⣿⣿⣿⣮⡚⣿⣿⣿⣿
⣿⡇⣿⣷⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⡼⣿⣿⣿
⣿⡷⢹⣿⣿⣿⣿⣿⣿⡿⢿⣿⣿⣿⣿⣿⣫⣝⢿⣿⣿⠹⣿⣿
⣿⡏⣾⣿⣿⣿⣿⣿⣿⠚⣖⢸⣿⠟⣛⣿⣤⣧⣾⣿⣿⣷⢹⣿
⣿⡇⣿⣿⣿⣿⣿⢿⣿⣿⣾⣿⣿⣷⣸⣿⣿⣿⣿⣯⢷⣿⢸⣿
⣿⣷⢸⣿⣿⣿⣽⢯⣷⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⢋⣾⣿
⣿⣿⣧⡫⡿⣿⣿⣿⣿⡿⠋⡉⠅⡉⠍⠛⢋⢡⠠⣐⠄⢻⣿⣿
⣿⣿⣿⣿⣶⣭⣍⢋⣫⠄⢐⢐⠐⡐⠈⠌⡀⡂⡐⢈⣭⣤⢻⣿
⣿⣿⡿⣟⠿⣿⡟⣾⣿⣿⣿⡿⠃⠄⠅⠡⠐⡀⢂⠌⠻⠿⣸⣿
⣿⣿⣷⣝⢗⡝⢡⣿⣿⣿⡉⠁⠡⠨⠠⠁⠅⢂⠂⠌⢐⣼⣿⣿
⣿⣿⣿⣿⡎⣷⢸⣿⣿⣿⣿⣶⣄⣄⡁⠌⠨⢀⣂⡅⣿⣿⣿⣿

LuckyBichar committed:
Merge remote-tracking branch 'origin/master' into bocharov.egor/S2
